core/src/main/java/com/klaytn/caver/tx/type/TxTypeFeeDelegatedAccountUpdate.java

```diff
@@ -81,27 +81,27 @@ public Type getType() {
      * @return TxTypeFeeDelegatedAccountUpdate decoded transaction
      */
     public static TxTypeFeeDelegatedAccountUpdate decodeFromRawTransaction(byte[] rawTransaction) {
-        byte[] rawTransactionExceptType = KlayTransactionUtils.getRawTransactionNoType(rawTransaction);
-
-        RlpList rlpList = RlpDecoder.decode(rawTransactionExceptType);
-        RlpList values = (RlpList) rlpList.getValues().get(0);
-
-        BigInteger nonce = ((RlpString) values.getValues().get(0)).asPositiveBigInteger();
-        BigInteger gasPrice = ((RlpString) values.getValues().get(1)).asPositiveBigInteger();
-        BigInteger gasLimit = ((RlpString) values.getValues().get(2)).asPositiveBigInteger();
-        String from = ((RlpString) values.getValues().get(3)).asString();
-        String accountkeyRaw = ((RlpString) values.getValues().get(4)).asString();
-
-        TxTypeFeeDelegatedAccountUpdate tx
-                = TxTypeFeeDelegatedAccountUpdate.createTransaction(nonce, gasPrice, gasLimit, from, AccountKeyDecoder.fromRlp(accountkeyRaw));
-        if (values.getValues().size() > 4) {
-            RlpList vrs = (RlpList) ((RlpList) (values.getValues().get(5))).getValues().get(0);
-            byte[] v = ((RlpString) vrs.getValues().get(0)).getBytes();
-            byte[] r = ((RlpString) vrs.getValues().get(1)).getBytes();
-            byte[] s = ((RlpString) vrs.getValues().get(2)).getBytes();
-            tx.setSenderSignatureData(new KlaySignatureData(v, r, s));
+        //TxHashRLP = type + encode([nonce, gasPrice, gas, from, rlpEncodedKey, txSignatures, feePayer, feePayerSignatures])
+        try {
+            byte[] rawTransactionExceptType = KlayTransactionUtils.getRawTransactionNoType(rawTransaction);
+
+            RlpList rlpList = RlpDecoder.decode(rawTransactionExceptType);
+            List<RlpType> values = ((RlpList) rlpList.getValues().get(0)).getValues();
+
+            BigInteger nonce = ((RlpString) values.get(0)).asPositiveBigInteger();
+            BigInteger gasPrice = ((RlpString) values.get(1)).asPositiveBigInteger();
+            BigInteger gasLimit = ((RlpString) values.get(2)).asPositiveBigInteger();
+            String from = ((RlpString) values.get(3)).asString();
+            String accountkeyRaw = ((RlpString) values.get(4)).asString();
+
+            TxTypeFeeDelegatedAccountUpdate tx
+                    = TxTypeFeeDelegatedAccountUpdate.createTransaction(nonce, gasPrice, gasLimit, from, AccountKeyDecoder.fromRlp(accountkeyRaw));
+            tx.addSignatureData(values, 5);
+
+            return tx;
+        } catch (Exception e) {
+            throw new RuntimeException("Incorrectly encoded tx.");
```

Comment:
I think this error message is not clear. how about this sentence? `There is a error in the processing of decoding tx`

Reply:
You are right.